sound/soc/sof/intel/hda-codec.c

```diff
@@ -37,7 +38,31 @@ static void hda_codec_load_module(struct hda_codec *codec)
 static void hda_codec_load_module(struct hda_codec *codec) {}
 #endif
 
+/* check jack status after resuming from suspend mode */
+void hda_codec_jack_check(struct snd_sof_dev *sdev, int status)
+{
+	struct hda_bus *hbus = sof_to_hbus(sdev);
+	struct hdac_bus *bus = sof_to_bus(sdev);
+	struct hda_codec *codec;
+
+	if (!status)
+		goto disable;
```

Comment:
I also don't think a `goto` is justified here. Maybe you can choose one of these solutions to eliminate it:
1. swap them
```
	/* disable controller Wake Up event*/
	snd_hdac_chip_writew(bus, WAKEEN,
			     snd_hdac_chip_readw(bus, WAKEEN) &
			     ~STATESTS_INT_MASK);

	if (!status)
		return;

	list_for_each_codec(...)
		...
```

2. sacrifice performance - typically you'll have 1 or two codecs, so, you can just remove that `if` and run the `list_for_each_codec(...)` loop directly, checking each status bit in the loop.

3. explicit `if`
```
	if (status)
		list_for_each_codec(codec, hbus) {
			...
```
IMHO either of those options is better than a goto.

Reply:
Thanks,  I have another patch for other feature and goto will be discarded.